internal/tjson/schema.go

```diff
@@ -123,15 +122,66 @@ var (
 )
 
 // Equal returns true if the schemas are equal.
+// For composite types schemas are equal if their types and subschemas are equal.
+// For scalar types schemas are equal if their types and formats are equal.
 func (s *Schema) Equal(other *Schema) bool {
 	if s == other {
 		return true
 	}
 
-	// TODO compare significant fields only (ignore title, description, etc.)
-	// TODO compare format according to type (for example, for Number, EmptyFormat == Double)
-	// https://github.com/FerretDB/FerretDB/issues/683
-	return reflect.DeepEqual(s, other)
+	if s.Type != other.Type {
+		return false
+	}
+
+	// If `s` and `other` are objects, compare their properties.
+	if s.Type == Object {
```

Comment:
Let's use `switch s.Type` there, as it is much more common in our code base

Reply:
Fixed